File: leetcode/1501-1800/1651-1700/1672.Richest Customer Wealth.cc
// https://leetcode.com/problems/richest-customer-wealth/

class Solution {
public:
    int maximumWealth(vector<vector<int>>& accounts) {
        int max_ = 0;
        
        int from = accounts[0].size() - 1;
        
        for (int i = from; i > 0; --i) {
            for (vector<int>& acc : accounts) {
                acc[0] += acc[i];
            }
        }
        
                    
        for (vector<int>& acc : accounts) {
            max_ = max(max_, acc[0]);
        }
        
        return max_;
    }
};

File: leetcode/1501-1800/1651-1700/1672.Richest Customer Wealth.rs
// https://leetcode.com/problems/richest-customer-wealth

impl Solution {
    pub fn maximum_wealth(accounts: Vec<Vec<i32>>) -> i32 {
        let mut max = 0;
        let wealths: Vec<i32> = accounts.iter().map(|customer| {
            customer.iter().fold(0, |acc, &account| account + acc)
        }).collect();

        for wealth in wealths {
            max = if max > wealth { max } else { wealth }
        }

        max
    }
}

impl Solution {
    pub fn maximum_wealth(accounts: Vec<Vec<i32>>) -> i32 {
        accounts.iter().map(|account| account.iter().sum()).max().unwrap()
    }
}